packages/account/src/Sections/Security/ClosingAccount/closing-account-reason.tsx

```diff
@@ -0,0 +1,142 @@
+                return (
+                    <ClosingAccountHasPendingConditions
+                        details={
+                            error &&
+                            typeof error === 'object' &&
+                            'details' in error &&
+                            typeof error.details === 'object'
+                                ? error.details
+                                : {}
+                        }
+                        onConfirm={redirectToSteps}
+                    />
+                );
+            case 'inaccessible_modal':
+                return (
+                    <ClosingAccountGeneralErrorContent
+                        message={
+                            error &&
+                            typeof error === 'object' &&
+                            'message' in error &&
+                            typeof error.message === 'string'
+                                ? error.message
+                                : ''
+                        }
+                        onClick={closeErrorModal}
+                    />
+                );
+            default:
+                return null;
+        }
+    };
+
+    const onConfirm = (final_reason: string) => {
+        setShowWarningModal(true);
+        setReasonsToCloseAccount(final_reason);
+    };
+
+    const closeErrorModal = () => {
+        setErrorInfo('');
+    };
+    const closeWarningModal = () => {
+        setShowWarningModal(false);
+    };
+
+    const startDeactivating = async () => {
+        closeWarningModal();
+        mutate({ payload: { reason: reasons_to_close_account } });
+    };
+
+    if (isSuccess) return <Redirect to={routes.account_closed} />;
+
+    if (isLoading) return <Loading is_fullscreen={false} />;
+
+    return (
+        <div className='closing-account-reasons'>
+            <Text weight='bold' size='xs' className='closing-account-reasons__title' as='p'>
+                <Localize
+                    i18n_default_text='Please tell us why you’re leaving. (Select up to {{ allowed_reasons }} reasons.)'
+                    values={{ allowed_reasons: MAX_ALLOWED_REASONS }}
+                />
+            </Text>
+            <ClosingAccountReasonForm onBackClick={redirectToSteps} onConfirmClick={onConfirm} />
+
+            {show_warning_modal && (
+                <ClosingAccountWarningModal
+                    show_warning_modal={show_warning_modal}
+                    closeWarningModal={closeWarningModal}
+                    startDeactivating={startDeactivating}
+                />
+            )}
+
+            {!!error_info && (
+                <Modal
+                    className='closing-account-reasons'
+                    is_open={!!error_info}
+                    toggleModal={closeErrorModal}
+                    title={getErrorModalTitle()}
+                >
+                    {getErrorModalContent()}
+                </Modal>
+            )}
```

Comment:
Isn't it fine to just pass `is_open` to display the modal rather than a condition to render in the DOM?

Reply:
i think this is better as we render the modal if only error is present

Reply:
Then don't you think `is_open` condition has no use? 🤔

Reply:
let me check the render

Reply:
updated @shahzaib-deriv